server/src/main/java/org/elasticsearch/search/aggregations/bucket/adjacency/AdjacencyMatrixAggregationBuilder.java

```diff
@@ -198,13 +198,13 @@ public String separator() {
     @Override
     protected AggregatorFactory doBuild(SearchContext context, AggregatorFactory parent, Builder subFactoriesBuilder)
             throws IOException {
-        int maxFilters = context.indexShard().indexSettings().getMaxAdjacencyMatrixFilters();
+        int maxFilters = SearchModule.INDICES_MAX_CLAUSE_COUNT_SETTING.get(context.indexShard().indexSettings().getSettings());
```

Comment:
I don't think this is correct since you are looking up a node setting in the index settings?

Reply:
Ah - good point.

Reply:
Presumably we just substitute for `BooleanQuery.getMaxClauseCount()` ?

Reply:
@jpountz I updated the PR to use BooleanQuery.getMaxClauseCount() as the source of truth.  Look good to you?